contracts/ConverterV2.sol

```diff
@@ -0,0 +1,198 @@
+    the underneath tokens in the LP pair (token0 and token1) to CNT. The CNT accumulated is used to allocate to different contracts
+    as per their allocation share.
+    pathForToken0 requires the path that will be used by the PolydexRouter to swap the token0 to CNT.
+    pathForToken1 requires the path that will be used by the PolydexRouter to swap the token1 to CNT.
+    */
+    function convertLP(address token0, address[] calldata pathForToken0, address token1, address[] calldata pathForToken1) external nonReentrant() {
+        // At least we try to make front-running harder to do.
+        require(msg.sender == tx.origin, "do not convert from contract");
+        IPolydexPair pair = IPolydexPair(factory.getPair(token0, token1));
+
+        require(address(pair) != address(0), "Invalid pair");
+
+        _safeTransfer(
+            address(pair),
+            address(pair),
+            pair.balanceOf(address(this))
+        );
+
+        pair.burn(address(this));
+        // swap everything to CNT
+        _swaptoCNT(token0, pathForToken0);
+        _swaptoCNT(token1, pathForToken1);
+        _allocateCNT();
+    }
+
+    /*
+    convertToken is used to convert ERC20 tokens received by the converter contract to CNT. It uses the PolydexRouter to convert
+    the ERC20 tokens to CNT. The CNT accumulated is used to allocate to different contracts as per their allocation share.
+    path param requires the path that will be used by the Router to swap the token to CNT.
+    */
+    function convertToken(address token, address[] calldata path) external nonReentrant() {
+        // At least we try to make front-running harder to do.
+        require(msg.sender == tx.origin, "do not convert from contract");
+        require(address(token) != address(0), "Invalid token address");
+        _swaptoCNT(token, path);
+        _allocateCNT();
+    }
+
+    /*
+    Internal method used by the converter to swap any token with the path for swapping to CNT with 
+    the help of PolydexRouter
+    */
+    function _swaptoCNT(address token, address[] calldata path) internal {
+        //the path should always have CNT otherwise it will convert to the token added which would lead to loss of funds.
+        uint amountIn = IERC20(token).balanceOf(address(this));
+        require(amountIn > 0, 'Contract should have token balance greater than 0');
+        require(IERC20(token).approve(address(router), amountIn), 'approve failed.');
+        uint amountOutMin = 1;
+        uint deadline = block.timestamp + 1;
+        router.swapExactTokensForTokens(amountIn, amountOutMin, path, address(this), deadline);
+    }
+
+    /*
+    Internal method used by the converter to allocate swapped/converted CNT 
+    to different contracts as per their allocation share.
+    */
+    function _allocateCNT() internal {
+        totalCNTAccumulated = IERC20(cnt).balanceOf(address(this));
```

Comment:
Do we need to set `totalCNTAllocated` variable in storage ? I think this can be ephemeral limited to only this function call. Anyways, we calculate the balance here.

Reply:
Makes sense :pray: